__tests__/app.test.js

```diff
@@ -346,7 +350,130 @@ describe('Articles', () => {
+            const commentBody = {
+				username: 'rogersop',
+                body: 'How might I try and break this API?'
+            };
+            return request(app)
+			.post(`/api/articles/${articleId}/comments`)
+			.send(commentBody)
+			.expect(400)
+            .then((response) => {
+                expect(response.body).toHaveProperty('msg');
+                expect(response.body.msg).toBe('bad request');
+            });
+        });
+
+        test('should respond with a 400 status (bad request) if the comment payload is invalid (user does not exist)', () => {
+            const articleId = 'this-is-not-a-valid-api-endpoint';
+            const commentBody = {
+				username: 'injecting-a-fake-user',
+                body: 'How might I try and break this API?'
+            };
+            return request(app)
+			.post(`/api/articles/${articleId}/comments`)
+			.send(commentBody)
+			.expect(400)
+            .then((response) => {
+                expect(response.body).toHaveProperty('msg');
+                expect(response.body.msg).toBe('bad request');
+            });
+        });
+
+        test('should respond with a 400 status (bad request) if the comment payload is invalid (body is not a string)', () => {
+            const articleId = 'this-is-not-a-valid-api-endpoint';
+            const commentBody = {
+				username: 'rogersop',
+                body: false
+            };
+            return request(app)
+			.post(`/api/articles/${articleId}/comments`)
+			.send(commentBody)
+			.expect(400)
+            .then((response) => {
+                expect(response.body).toHaveProperty('msg');
+                expect(response.body.msg).toBe('bad request');
+            });
+        });
+
+        test('should respond with a 400 status (bad request) if the comment payload does not contain a username', () => {
+            const articleId = 'this-is-not-a-valid-api-endpoint';
+            const commentBody = {
+                body: 'How might I try and break this API?'
+            };
+            return request(app)
+			.post(`/api/articles/${articleId}/comments`)
+			.send(commentBody)
+			.expect(400)
+            .then((response) => {
+                expect(response.body).toHaveProperty('msg');
+                expect(response.body.msg).toBe('bad request');
+            });
+        });
+
+        test('should respond with a 400 status (bad request) if the comment payload does not contain a body', () => {
+            const articleId = 'this-is-not-a-valid-api-endpoint';
+            const commentBody = {
+                username: 'rogersop'
+            };
+            return request(app)
+			.post(`/api/articles/${articleId}/comments`)
+			.send(commentBody)
+			.expect(400)
+            .then((response) => {
+                expect(response.body).toHaveProperty('msg');
+                expect(response.body.msg).toBe('bad request');
+            });
+        });
+
+    });
+
+});
```

Comment:
It would be good to see a 201 if unnecessary properties are added (post but ignore them)